### PR TITLE
Instructor: recycle bin feature: refactor term 'recovery' to other meaningful terms #9013

### LINKED ISSUE #9013: Instructor: recycle bin feature: refactor term 'recovery' to other meaningful terms
**Environment:**
`recycling-bin` branch at commit https://github.com/TEAMMATES/teammates/commit/a88eb1caba3fcdeb748d643d3c8288f8ab2839a6

**Examples:**
- `getRecoveryCoursesForInstructors` -> `getSoftDeletedCoursesForInstructors`
- `recoveryCoursesTable` -> `softDeletedCoursesTable`
- `restoreFeedbackSessionFromRecovery` -> `restoreFeedbackSessionFromRecycleBin`

### PR DESCRIPTION
**Pull request by dalessr**
Fixes #9013

### CODE REVIEW

**Comment by dalessr** [comment]
@darrenwee @joanneong Ready for review once #8969 has been merged :)

**Comment by wkurniawan07** [comment]
@dalessr Do you really need all of these to be done on the `recycling-bin` branch? The ones already in the current `master` branch, you can make a PR directly to `master`.

**Comment by dalessr** [comment]
@wkurniawan07 Oh because I was doing it for both courses and sessions so that it won't cause conflicts with #8969 . Otherwise it will be very troublesome when merging #8969 .

**Comment by dalessr** [comment]
@darrenwee @joanneong Ready for review

File: src/main/java/teammates/ui/template/SoftDeletedFeedbackSessionsTable.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package teammates.ui.template;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SoftDeletedFeedbackSessionsTable {
```
Comment (darrenwee) [inline src/main/java/teammates/ui/template/SoftDeletedFeedbackSessionsTable.java]: What is this for?

Comment (dalessr) [reply src/main/java/teammates/ui/template/SoftDeletedFeedbackSessionsTable.java]: This is a table that contains information of soft-deleted feedback sessions in the Recycle Bin. I renamed the file from `RecoveryFeedbackSessionsTable` to `SoftDeletedFeedbackSessionsTable`, so in `Files changed` page it shows that I deleted a file & created a new file.

**Pull request merged**